zerver/lib/streams.py

```diff
@@ -52,7 +52,7 @@ def access_stream_common(user_profile, stream, error):
         sub = None
 
     # If the stream is in your realm and public, you can access it.
-    if stream.is_public():
+    if stream.has_public_history:
         return (recipient, sub)
```

Comment:
This one we may want to be careful about; since I think we use this access function for joining as well.

Reply:
Maybe leave this one on `.is_public()` since I think disentangling here will be complex.